### ISSUE TITLE
[🙏] IaC mode lacks the --policy-path flag

### ISSUE THREAD

**Issue by Retler** [snyk/snyk#1786]
- IaC scan mode 

**Describe the user need**
I want Snyk to be able to use the --policy-path when  running `snyk iac test`.

**Describe expected behaviour**
`snyk iac test --policy-path=path/to/.snyk` should work and use the policy in `path/to/.snyk`.

**Additional context**
In my particular case, there are some k8s signatures I want to ignore.

**Comment by Retler** [comment]
Hi @rontalx and thanks for the answer. This indeed helps us a bit, but we have cases where we want to ignore rules for specific kubernetes files or "paths" (i.e. an Nginx deployment).

We will use the UI for now. I hope some more flexibility will come for IaC scans. 

By the way, thanks for an awesome piece of software!

**Comment by rontalx** [comment]
Thanks for the great feedback @Retler! 
More granular ignores are definitely something we are considering in the near future,
so feel free to share any potential use-cases you had in mind.

**Comment by Retler** [comment]
Where is the right forum to share such use-cases and is there any formal template we must follow?

**Comment by rontalx** [comment]
currently these are the available options:
Freemium users:  either Github Issues or [the support portal](https://support.snyk.io/), we're potentially checking opening GH Discussion for the future, currently it's disabled.
Paid users: can use these mentioned above + an internal slack channel for customers.

No formal template at the moment.
Thanks for raising this.

**Comment by craigfurman** [comment]
@Retler the first iteration of IaC support for ignore policies just landed in the Snyk CLI version released this morning. Docs: https://support.snyk.io/hc/en-us/articles/4405381463313-IaC-ignores-using-the-snyk-policy-file.

**Comment by pecigonzalo** [comment]
I cant seem to get this to work
```
❯ snyk --version     

1.695.0 (standalone)

❯ cat .snyk 
.snyk 
version: v1.19.0
ignore:
  SNYK-CC-TF-55:
    - '*':
        reason: CloudWatch Alerts do not require KMS

❯ snyk iac test environments/alerts.tf           


Testing environments/alerts.tf...


Infrastructure as code issues:
  ✗ Non-Encrypted SNS Topic [Medium Severity] [SNYK-CC-TF-55] in SNS
    introduced by resource > aws_sns_topic[cloudwatch_alerts] > kms_master_key_id

  ✗ SNS topic is not encrypted with customer managed key [Low Severity] [SNYK-CC-AWS-422] in SNS
    introduced by resource > aws_sns_topic[cloudwatch_alerts] > kms_master_key_id


Organization:      foo
Type:              Terraform
Target file:       environments/alerts.tf
Project name:      environments
Open source:       no
Project path:      environments/alerts.tf

Tested environments/alerts.tf for known issues, found 2 issues
```

**Comment by craigfurman** [comment]
Hi @pecigonzalo, file arguments to `snyk iac test` (as opposed to directory arguments) are somewhat of a special case for `.snyk` ignores: you must pass the policy file path in, it will not be auto-detected. See https://support.snyk.io/hc/en-us/articles/4405381463313-IaC-ignores-using-the-snyk-policy-file, "Policy flags and policy file notes" section.

`snyk iac test --policy-path=$PWD/.snyk environments/alerts.tf` should work as you expect it to.

**Comment by pecigonzalo** [comment]
@craigfurman Thanks, that worked as expected now. I got confused because there is another bug (already reported) in which a file like (not the same I shared before):

```
version: v1.19.0
ignore:
  SNYK-CC-TF-55:
    - 'environments/alerts.tf':
        reason: CloudWatch Alerts do not require KMS
```

would not correctly ignore `SNYK-CC-TF-55` in `environments/alerts.tf`. I must have been using this other file when I tried using `--policy-path=` and it did not work because of this other unrelated issue.

TLDR: I think I got hit by the 2 situations at once and made my local testing more confusing.

Thanks, that indeed works if the policy works.

**Comment by craigfurman** [comment]
@pecigonzalo no worries, glad this worked!

I think for your second case, you need to use glob-ish syntax (https://support.snyk.io/hc/en-us/articles/4405381463313-IaC-ignores-using-the-snyk-policy-file):

```
SNYK-CC-TF-55:
    - 'environments/alerts.tf > *':
```

Does that work?

**Comment by pecigonzalo** [comment]
Yeah that works that is what I reported in the other bug report.
